gst/nnstreamer/tensor_query/tensor_query_common.c

```diff
@@ -790,23 +789,27 @@ _message_handler (void *thread_data)
     }
 
     if (result || gst_caps_can_intersect (client_caps, server_caps)) {
-      cmd_data.cmd = _TENSOR_QUERY_CMD_RESPOND_APPROVE;
-      cmd_data.data.data = (uint8_t *) _sdata->sink_caps_str;
-      cmd_data.data.size = (size_t) strlen (_sdata->sink_caps_str) + 1;
+      cmd_data_send.cmd = _TENSOR_QUERY_CMD_RESPOND_APPROVE;
+      cmd_data_send.data.data = (uint8_t *) _sdata->sink_caps_str;
+      cmd_data_send.data.size = (size_t) strlen (_sdata->sink_caps_str) + 1;
     } else {
       /* respond deny with src caps string */
       nns_loge ("Query caps is not acceptable!");
-      nns_loge ("Query client sink caps: %s", (char *) cmd_data.data.data);
+      nns_loge ("Query client sink caps: %s", cmd_data_receive.data.data);
       nns_loge ("Query server src caps: %s", _sdata->src_caps_str);
-      cmd_data.cmd = _TENSOR_QUERY_CMD_RESPOND_DENY;
-      cmd_data.data.data = (uint8_t *) _sdata->src_caps_str;
-      cmd_data.data.size = (size_t) strlen (_sdata->src_caps_str) + 1;
+
+      cmd_data_send.cmd = _TENSOR_QUERY_CMD_RESPOND_DENY;
+      cmd_data_send.data.data = (uint8_t *) _sdata->src_caps_str;
+      cmd_data_send.data.size = (size_t) strlen (_sdata->src_caps_str) + 1;
     }
 
+    free (cmd_data_receive.data.data);
```

Comment:
We use glib functions. Use g_free() instead of free().

Reply:
It seems that this buffer was allocated by `malloc`. This should be also fixed.
https://github.com/nnstreamer/nnstreamer/blob/8155d7944b4520b080399c79e82192f7a5d03dec/gst/nnstreamer/tensor_query/tensor_query_common.c#L355-L357

Reply:
@niley7464 please check the above comment.

Reply:
I fixed both which are not using glib functions! Thank you